### ISSUE TITLE
Browser locks up on extremely long pages

### ISSUE THREAD

**Issue by AufbeschissenerKunde** [Quicksaver/FindBar-Tweak#258]
Here's a fun test case for you:  [http://faculty.washington.edu/rturner1/AustraLearn_GBR.mht](http://faculty.washington.edu/rturner1/AustraLearn_GBR.mht)

This page is nearly 1500 pages long in print-preview.  (I looked just to get that figure.)  When I try to search this page, the browser totally locks up, is marked as "not responding," and turns pale.  I'm not really sure how long it gets stuck like this because I get impatient and kill the process.

If I disable FindBar, the regular built-in search works without issue.

I acknowledge this is probably my own fault for trying to use an enhanced search with fancy visuals, but we can also look at this as an opportunity to figure out optimization for extremely long pages.

**Comment by Quicksaver** [comment]
Yeah, it has a little trouble handling huge pages like that. Improving that is already on my to-do list, although I don't know when I'll be able to do it.

BTW it's probably not the page being huge, but having a lot of matches, as in the thousands. Usually the add-on handles well a few matches even in long pages like that.

**Comment by FatOrangutan** [comment]
I would like to add my vote to this. On a page such as [this one](http://pastebin.com/6NJpiE78), the native FF search for, say, `g` takes place instantly, while FBT, even with 'Highlight all matches by default', 'Show highlights grid in the scrollbar' and 'Show sights on current hit / all hits' disabled - which should mean it is doing nothing more than the native search - brings the browser to a halt.  
  
What's worse, if I type `greasemonkey` right away, FBT should directly search for the whole word instead of starting its search with `g` which is gonna lock up the browser.

**Comment by mausalimi** [comment]
well, if i start typing "greasemonkey" fast enough, it works.
if not, yes, the browser is locked for a very long time, eating the whole CPU core.
